packages/web/server/queries/complex/earn/strategies.ts

```diff
@@ -0,0 +1,242 @@
+import { Dec, PricePretty, RatePretty } from "@keplr-wallet/unit";
+import cachified, { CacheEntry } from "cachified";
+import dayjs from "dayjs";
+import { LRUCache } from "lru-cache";
+
+import { type Asset, getAsset } from "~/server/queries/complex/assets";
+import { DEFAULT_VS_CURRENCY } from "~/server/queries/complex/assets/config";
+import {
+  EarnStrategyBalance,
+  queryEarnUserBalance,
+  queryStrategyAPR,
+  queryStrategyTVL,
+  RawStrategyCMSData,
+  RawStrategyTVL,
+  StrategyAnnualPercentages,
+  StrategyCMSData,
+  StrategyTVL,
+} from "~/server/queries/numia/earn";
+import { queryOsmosisCMS } from "~/server/queries/osmosis/cms";
+import { DEFAULT_LRU_OPTIONS } from "~/utils/cache";
+
+const earnStrategyBalanceCache = new LRUCache<string, CacheEntry>(
+  DEFAULT_LRU_OPTIONS
+);
+
+const earnStrategyAnnualPercentagesCache = new LRUCache<string, CacheEntry>(
+  DEFAULT_LRU_OPTIONS
+);
+
+const earnStrategyTVLCache = new LRUCache<string, CacheEntry>(
+  DEFAULT_LRU_OPTIONS
+);
+
+const earnRawStrategyCMSDataCache = new LRUCache<string, CacheEntry>(
+  DEFAULT_LRU_OPTIONS
+);
+export async function getStrategyBalance(
+  strategyId: string,
+  userOsmoAddress: string
+) {
+  return await cachified({
+    cache: earnStrategyBalanceCache,
+    ttl: 1000 * 20,
+    key: `earn-strategy-balance-${strategyId}-${userOsmoAddress}`,
+    getFreshValue: async (): Promise<EarnStrategyBalance | undefined> => {
+      try {
+        const { balance, strategy, unclaimed_rewards } =
+          await queryEarnUserBalance(strategyId, userOsmoAddress);
+
+        return {
+          balance: {
+            amount: balance.amount,
+            usd: new PricePretty(DEFAULT_VS_CURRENCY, new Dec(balance.usd)),
+          },
+          id: strategy,
+          unclaimed_rewards: {
+            usd: new PricePretty(
+              DEFAULT_VS_CURRENCY,
+              new Dec(unclaimed_rewards.total_usd)
+            ),
+          },
+        };
+      } catch (error) {
+        throw new Error("Error while fetching strategy balance", {
+          cause: error as Error,
+        });
+      }
+    },
+  });
+}
+
+export async function getStrategyAnnualPercentages(strategyId: string) {
+  return await cachified({
+    cache: earnStrategyAnnualPercentagesCache,
+    ttl: 1000 * 20,
+    key: `earn-strategy-annualPercentages-${strategyId}`,
+    getFreshValue: async (): Promise<StrategyAnnualPercentages> => {
+      try {
+        const { apr } = await queryStrategyAPR(strategyId);
+        return {
+          apr: new RatePretty(apr),
+          apy: new RatePretty(calculateAPY(apr)),
+        };
+      } catch (error) {
+        throw new Error("Error while fetching strategy APR", {
+          cause: error as Error,
+        });
```

Comment:
Are each of these catch and throws necessary? We could just let the exception bubble from the query Fns.